stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/interceptor/internal/SendMessageInterceptorImpl.kt

```diff
@@ -232,21 +233,25 @@ internal class SendMessageInterceptorImpl(
             Result.success(messageToBeSent.copy(type = Message.TYPE_REGULAR))
         } else {
             Result.error(ChatError("Could not upload attachments, not sending message with id ${newMessage.id}"))
+        }.also {
+            uploadIds.remove(newMessage.id)
```

Comment:
I believe we need to remove the id from `uploadIds` only if the result is successful. Otherwise, the worker will still exist and retry sending the attachment and we won't be able to cancel it, if needed

Reply:
I think this `Result.error` is not from worker but when all attachments of this message have `upload failed state`. That should happen once a worker has finished his job. So it doesn't seem right to keep this work id stored.

Reply:
Based on `UploadAttachmentsAndroidWorker` & `UploadAttachmentsWorker` it seems like worker waits until all attachments get processed and it fails if one of those attachment got failed.

It looks like we're good to remove an `uploadId` here in both cases